dockerfiles/alpine/Dockerfile

```diff
@@ -0,0 +1,74 @@
+FROM rust:1.72 as builder
+
+ENV HOME=/home/root
+WORKDIR $HOME/app
+# rust toolchain
+ARG RUST_TC_VER=stable
+ARG BUILD_TARGET=x86_64-unknown-linux-musl
+ARG BUILD_PROFILE=release-lto
+
+# import the Nodesource GPG key
+RUN apt-get update \
+    && apt-get install -y ca-certificates curl gnupg \
+    && mkdir -p /etc/apt/keyrings \
+    && curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+
+# Create deb repository
+# FIXME: Replace it with Nodejs 20.x when it will be fixed
+ENV NODE_MAJOR=18
+RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+
+# Install yarn keys
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \ 
+    && echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+
+RUN apt-get update \
+  && apt-get install -y --no-install-recommends musl-tools yarn nodejs \
+  && rustup install $RUST_TC_VER \
+  && rustup default $RUST_TC_VER \
+  && rustup target add $BUILD_TARGET
+
+
+# RUN USER=root cargo new --bin bob-management
+COPY ./Cargo.toml ./Cargo.toml
+COPY ./build.rs ./build.rs
+COPY ./cli ./cli
+COPY ./backend ./backend
+COPY ./frontend ./frontend
+RUN cargo build --profile=$BUILD_PROFILE --target=$BUILD_TARGET --bin bob_management
+RUN find . -name '*.rs' -delete
+
+ADD . ./
+
+RUN rm ./target/$BUILD_TARGET/$BUILD_PROFILE/deps/bob_management* 
+
+RUN --mount=type=cache,target=/usr/local/cargo/registry \
+    --mount=type=cache,target=/home/root/app/target \
+    cargo build --profile=$BUILD_PROFILE --target=$BUILD_TARGET --bin bob_management \
```

Comment:
Why do you rebuild everything again?

If you use `Dockerfile` from Bob as an example, then there in the first phase of the build only `Cargo.toml` files are copied and fake `main.rs` and `lib.rs` are created, and only in the second phase the entire code is copied and built. This is done in order to download and build only dependencies in the first phase. The layer with the built dependencies will remain in the Docker cache, which means that the following builds without changing of the list of dependencies will use this cache and be done much faster

Reply:
yeah, I tried to replicate that behavior, but failed to do that properly, thanks for heads up